whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapService.java

```
@@ -174,7 +174,45 @@ public Response searchEntities(
             return handleSearch(new String[]{"organisation", "nic-hdl"}, handle, request);
         }
 
-        throw new RdapException("400 Bad Request", "The server is not able to process the request", HttpStatus.BAD_REQUEST_400);
+        throw new RdapException("400 Bad Request", "Either fn or handle is a required parameter, but never both", HttpStatus.BAD_REQUEST_400);
+    }
+
+    @GET
+    @Produces({MediaType.APPLICATION_JSON, CONTENT_TYPE_RDAP_JSON})
+    @Path("/ips")
+    public Response searchIps(
+            @Context final HttpServletRequest request,
+            @QueryParam("name") final String name,
+            @QueryParam("handle") final String handle) {
+
+        LOGGER.debug("Request: {}", RestServiceHelper.getRequestURI(request));
+
+        if (name != null && handle == null || name == null && handle != null) {
```

fail early

---

it is not possible to apply fail early here